server/paymentProviders/braintree/webhooks.ts

```diff
@@ -0,0 +1,91 @@
+import OrderStatus from '../../constants/order_status';
+import logger from '../../lib/logger';
+import { sendThankYouEmail } from '../../lib/recurring-contributions';
+import models from '../../models';
+
+import { getBraintreeGatewayForHost } from './gateway';
+import { BraintreeTransactionAlreadyExistsError, createTransactionsPairFromBraintreeTransaction } from './helpers';
+
+const onSubscriptionChargedSuccessfully = async (braintreeNotification): Promise<void> => {
+  // Ignore the first event, as it's recorded directly in `processOrder`
+  if (braintreeNotification.subscription.currentBillingCycle === 1) {
+    logger.debug('Braintree notification is for the first charge, ignoring');
+    return;
+  }
+
+  // These cases are not suppose to happen, but we add some logs just in case
+  const braintreeTransactions = braintreeNotification.subscription.transactions;
+  if (!braintreeTransactions.length) {
+    logger.warn(`No transactions provided for charged subscription ${braintreeNotification.subscription.id}`);
+  } else if (braintreeTransactions.length > 1) {
+    logger.warn(`Got multiple transactions for charged subscription ${braintreeNotification.subscription.id}`);
+  }
+
+  // Retrieve the order and do some sanity checks to make sure the subscription is healthy
+  const order = await models.Order.findOne({
+    include: [
+      { association: 'paymentMethod', required: true },
+      { association: 'collective' },
+      { association: 'fromCollective' },
+      { association: 'createdByUser' },
+    ],
+    where: {
+      data: {
+        braintree: {
+          subscriptionId: braintreeNotification.subscription.id, // TODO(Braintree): Create an index for this
+        },
+      },
+    },
+  });
+
+  if (!order) {
+    throw new Error(`No order found for subscription: ${braintreeNotification.subscription.id}`);
+  } else if (!order.collective) {
+    throw new Error(
+      `Received a subscription payment (order #${order.id}) for a deleted collective. Please unsubscribe immediately and refund the transaction!`,
+    );
+  } else if (!order.collective.isActive) {
+    throw new Error(
+      `Received a subscription payment (order #${order.id}) for an inactive collective. Please unsubscribe immediately and refund the transaction!`,
+    );
+  }
+
+  // Record the transaction
+  let transaction;
+  try {
+    for (const braintreeTransaction of braintreeTransactions) {
+      transaction = await createTransactionsPairFromBraintreeTransaction(order, braintreeTransaction);
```

Comment:
Thanks for catching that, I actually misunderstood this field. According to https://developers.braintreepayments.com/reference/response/subscription/node#transactions: 

> Transactions associated with the subscription, sorted by creation date with the most recent first. When returned in a subscription webhook, this array contains only the 20 most recent transactions; otherwise, it contains all transactions associated with the subscription.

So we should only look at `braintreeNotification.subscription.transactions[0];`. I'll update the code.